pkg/actor/upgrade.go

```diff
@@ -0,0 +1,192 @@
+package actor
+
+import (
+	"context"
+	"encoding/csv"
+	"fmt"
+	"github.com/Masterminds/semver/v3"
+	api "github.com/cockroachlabs/crdb-operator/api/v1alpha1"
+	"github.com/cockroachlabs/crdb-operator/pkg/condition"
+	"github.com/cockroachlabs/crdb-operator/pkg/kube"
+	"github.com/cockroachlabs/crdb-operator/pkg/resource"
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kubetypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+)
+
+func newUpgrade(scheme *runtime.Scheme, cl client.Client, config *rest.Config) Actor {
+	return &upgrade{
+		action: newAction("upgrade", scheme, cl),
+		config: config,
+	}
+}
+
+type upgrade struct {
+	action
+
+	config *rest.Config
+}
+
+func (up *upgrade) Handles(conds []api.ClusterCondition) bool {
+	return condition.False(api.NotInitializedCondition, conds)
+}
+
+func (up *upgrade) Act(ctx context.Context, cluster *resource.Cluster) error {
```

Comment:
Finalizing the release is done outside the operator? Any plans to change this?

Reply:
Yes, currently, I think it is better to leave to some external party (a human or an automation) to finalize the update. Later once we have more feedback, we can revise it. The main problem here as it appears to me is how the operator can verify that it is fine to finalize as it does not have feedback from the applications using the database.